quic/transport/ngtcp2/connection/disconnectingstate.nim

```diff
@@ -0,0 +1,51 @@
+import pkg/chronos
+import ../../../udp/datagram
+import ../../quicconnection
+import ../../connectionid
+import ../../stream
+import ./closedstate
+
+type
+  DisconnectingConnection* = ref object of ConnectionState
+    connection: QuicConnection
+    disconnect: Future[void]
+    ids: seq[ConnectionId]
+
+proc newDisconnectingConnection*(ids: seq[ConnectionId]):
+                                            DisconnectingConnection =
+  DisconnectingConnection(ids: ids)
+
+proc callDisconnect(connection: QuicConnection) {.async.} =
+  if connection.disconnect != nil:
+    await connection.disconnect()
+
+{.push locks: "unknown".}
```

Comment:
What is this working around?

Reply:
I ran into warnings about the lock level (ab4fa24). From the [Nim documentation](https://nim-lang.github.io/Nim/manual_experimental.html#guards-and-locks-lock-levels):

The locks pragma can also take the special value "unknown". This is useful in the context of dynamic method dispatching. In the following example, the compiler can infer a lock level of 0 for the base case. However, one of the overloaded methods calls a procvar which is potentially locking. Thus, the lock level of calling g.testMethod cannot be inferred statically, leading to compiler warnings. By using {.locks: "unknown".}, the base method can be marked explicitly as having unknown lock level as well